### PR TITLE
test(website): add AST viewer test case

### LINKED ISSUE #6086: Repo: Add an end-to-end (Playwright) test for the playground's AST viewer
### Suggestion

Following #6080, I realized there was one more test for the https://typescript-eslint.io/play playground that I wanted to add!

1. Type some code into the editor
2. Enable the AST viewer
3. Check that the AST shows nodes corresponding to the code

Leaving this open as a `good first issue` so anybody who wants to contribute can. 

Screenshot of typescript-eslint.io showing the ESTree AST for console.log('hello, world!')

### PR DESCRIPTION
**Pull request by tapanprakasht**
## PR Checklist

- [x] Addresses an existing open issue: fixes #6086
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

## Overview

Added AST viewer playwright test case.

### CODE REVIEW

**Review by JoshuaKGoldberg — APPROVED** [review]
💯, thanks @tapanprakasht! This is great.

**Comment by tapanprakasht** [comment]
@JoshuaKGoldberg Thanks for checking the PR. I do face some challenges here for the test case. `writeInEditor` doesn't seems to be behaving as expected. I see typed text in the editor disappear automatically while running the test case which is causing the test to fail.

File: packages/website/tests/playground.spec.ts [hunk 1]
```diff
@@ -46,6 +46,19 @@ test.describe('Playground', () => {
...
+      .getByRole('combobox', { name: 'AST Viewer' })
+      .selectOption({ label: 'ESTree' });
+    // 3. Validate variable declaration block exists in AST viewer
+    await expect(
+      page.getByRole('link', { name: 'VariableDeclaration' }),
+    ).toBeVisible();
+  });
 });
 
 async function writeInEditor(page: Page, text: string): Promise<void> {
```
Comment (JoshuaKGoldberg) [inline packages/website/tests/playground.spec.ts:64]: Threading https://github.com/typescript-eslint/typescript-eslint/pull/6275#issuecomment-1365289946: hmm, yeah, I see what you mean. When I run it locally, it looks like Monaco is wiping out the typed text:

1. The test selects & deletes all text
2. The test types in new text
3. ...some hundreds of milliseconds later...
4. Monaco deletes all the newly added text

My first instinct for a workaround is to wait until Monaco is done processing whatever it needs to do. Vaguely:

1. Type some bogus text into the editor
2. Wait until the text dissapears
3. ...then do the rest of the text

I prototyped this locally and it worked out well. What do you think?

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Fun!

File: packages/website/tests/playground.spec.ts (hunk 1 above)
Comment (tapanprakasht) [reply packages/website/tests/playground.spec.ts:76]: Hi @JoshuaKGoldberg, Workaround looks good. Meanwhile, I was checking another workaround by looking at the previous test case. I tried out something like, opening the lint editor and going back to the code editor again. Then the typed text persisted without wiping out. It worked locally. Modified the current test based on it.

Comment (armano2) [reply packages/website/tests/playground.spec.ts:76]: > My first instinct for a workaround is to wait until Monaco is done processing whatever it needs to do. Vaguely:

workaround should not be needed, someone reported similar issue in browser, we should fix this bug

Comment (armano2) [reply packages/website/tests/playground.spec.ts:76]: this issue should be fixed now (#6513), as long as you focus editor content is not going to be cleared anymore :)

Comment (armano2) [reply packages/website/tests/playground.spec.ts:76]: to focus editor and wait for it you have to do something like this:

```
  await page.isVisible('.monaco-editor .monaco-mouse-cursor-text');
  const monacoEditor = page
    .locator('.monaco-editor .monaco-mouse-cursor-text')
    .nth(0);
  await monacoEditor.focus();
```

and as for this test, you can do something like this:

```
    // 1. Enable AST viewer
    await page
      .getByRole('combobox', { name: 'AST Viewer' })
      .selectOption({ label: 'ESTree' });

    // 2. Type some valid code in the playground
    await writeInEditor(page, 'let value: Array<string>;');

    // 3. Wait for the AST viewer to update
    await page.waitForEvent(
      'console',
      message => message.text() === '[Editor] linting triggered',
    );

    // 4. Validate variable declaration block exists in AST viewer
    await expect(
      page.getByRole('link', { name: 'VariableDeclaration' }),
    ).toBeVisible();
```

**Comment by tapanprakasht** [comment]
> Oh hey thanks for the reminder armano2 - ping @tapanprakasht, if you merge the latest `main` into this branch I think you should be unblocked. Is there anything you're waiting for our input on?

@JoshuaKGoldberg Updated the test cases on top the latest master branch

**Review by JoshuaKGoldberg — APPROVED** [review]
Wonderful, looks like all is passing! Thanks @tapanprakasht - this is great. Sorry it took so long to get it approved!

**Pull request merged**